packages/core/test-helpers/src/websocket.ts

```diff
@@ -0,0 +1,65 @@
+import { WebSocketClassProvider } from '@chainlink/ea-bootstrap/src/lib/ws/recorder'
+import { Server, WebSocket } from 'mock-socket'
+
+/**
+ * Sets the mocked websocket instance in the provided provider class.
+ * We need this here, because the tests will connect using their instance of WebSocketClassProvider;
+ * fetching from this library to the @chainlink/ea-bootstrap package would access _another_ instance
+ * of the same constructor. Although it should be a singleton, dependencies are different so that
+ * means that the static classes themselves are also different.
+ *
+ * @param provider singleton WebSocketClassProvider
+ */
+export const mockWebSocketProvider = (provider: typeof WebSocketClassProvider): void => {
+  // Need to disable typing, the mock-socket impl does not implement the ws interface fully
+  provider.set(WebSocket as any) // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Could you do something like this?
```
import { Server, WebSocket as MockWebSocket } from 'mock-socket'
import WebSocket from 'ws'
...
provider.set(MockWebSocket as WebSocket)
```
We really shouldn't be using `any` moving forward.

Reply:
Unfortunately the typing issue lies in the incompatibility of the types specified by the rxjs library and the Node websocket, seen originally [here](https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/core/bootstrap/src/lib/ws/epics.ts#L183). Additionally, the WebSocket impl from the mock not being a full impl from the WebSocket interface, makes it so again the types have to be manually coerced. It's a bit of an unfortunate situation, because ideally all libraries would have correct types, but it's not up to us in this case.

Reply:
Yeah, makes sense. I guess it's only for the mock testing, so it's not too bad...